src/renderers/shared/fiber/ReactFiberReconciler.js

```diff
@@ -170,7 +170,7 @@ module.exports = function<T, P, I, TI, PI, C, CX>(config : HostConfig<T, P, I, T
 
     deferredUpdates,
 
-    getPublicRootInstance(container : OpaqueNode) : (ReactComponent<any, any, any> | I | TI | null) {
+    getPublicRootInstance(container : OpaqueNode) : (ReactComponent<any, any, any> | PI | null) {
```

Comment:
The more I think about this the less I’m convinced that this should be changed from `I | TI` to `PI`. `PI` was only added for ` getPublicInstance` which is only called via `attachRef`. The naming may be unfortunate since `getHostPublicInstance` and `getPublicInstance` aren’t related (internally).

Fwiw, I’m not attempting or desiring to implement anything new here, just trying to understand what the behavior was before #8628.